docs/data-model/zed.md

```diff
@@ -0,0 +1,227 @@
+
+There are 30 types of primitive values with syntax defined as follows:
+
+| Name       | Definition                                      |
+|------------|-------------------------------------------------|
+
+| `uint8`    | unsigned 8-bit integer  |
+| `uint16`   | unsigned 16-bit integer |
+| `uint32`   | unsigned 32-bit integer |
+| `uint64`   | unsigned 64-bit integer |
+| `uint128`  | unsigned 128-bit integer |
+| `uint256`  | unsigned 256-bit integer |
+| `int8`     | signed 8-bit integer    |
+| `int16`    | signed 16-bit integer   |
+| `int32`    | signed 32-bit integer   |
+| `int64`    | signed 64-bit integer   |
+| `int128`   | signed 128-bit integer   |
+| `int256`   | signed 256-bit integer   |
+| `duration` | signed 64-bit integer as nanoseconds |
+| `time`     | signed 64-bit integer as nanoseconds from epoch |
+| `float16`  | IEEE-754 binary16 |
+| `float32`  | IEEE-754 binary32 |
+| `float64`  | IEEE-754 binary64 |
+| `float128`  | IEEE-754 binary128 |
+| `float256`  | IEEE-754 binary256 |
+| `decimal32`  | IEEE-754 decimal32 |
+| `decimal64`  | IEEE-754 decimal64 |
+| `decimal128`  | IEEE-754 decimal128 |
+| `decimal256`  | IEEE-754 decimal256 |
+| `bool`     | the Boolean value `true` or `false` |
+| `bytes`    | a bounded sequence of 8-bit bytes |
+| `string`   | a UTF-8 string |
+| `ip`       | an IPv4 or IPv6 address |
+| `net`      | an IPv4 or IPv6 address and net mask |
+| `type`     | a Zed type value |
+| `null`     | the null type |
+
+> Note that `time` values correspond to 64-bit epoch nanoseconds and thus
+> not all possible RFC 3339 date/times are valid.  In addition,
+> nanosecond epoch times overflow on April 11, 2262.
+> For the world of 2262, a new epoch can be created well in advance
+> and the old time epoch and new time epoch can live side by side with
+> the old using a named type for the new epoch time aliased to the old `time`.
+> An app that wants more than 64 bits of timestamp precision can always use
+> a named type of a `bytes` type and do its own conversions to and from the
+> corresponding bytes values.  A time with a local time zone can be represented
+> as a Zed record of a time field and a zone field
+
+## 2. Complex Types
+
+Complex types are composed of primitive types and/or other complex types.
+The _classes_ of complex types include:
+* _record_ - an ordered collection of zero or more named values called fields,
+* _array_ - an ordered sequence of zero or more values called elements,
+* _set_ - a set of zero or more unique values called elements,
+* _union_ - a type representing values whose type is any of a specified collection of two or more unique types,
+* _enum_ - a type representing a finite set of symbols typically representing categories,
+* _map_ - a collection of zero or more key/value pairs where the keys are of a
+uniform type called the key type and the values are of a uniform type called
+the value type, and
+* _error_ - any value wrapped as an "error".
```

Comment:
I think grouping the collection types together would make this list a little easier to digest. How do you feel about moving map up next to its buddies when the next flag day comes?

Reply:
Good idea!